clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -186,9 +187,15 @@ public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
      */
     public CompletableFuture<Map<TopicPartition, OffsetAndTimestampInternal>> fetchOffsets(
             Map<TopicPartition, Long> timestampsToSearch,
-            boolean requireTimestamps) {
+            boolean requireTimestamps,
+            List<CompletableFuture<RuntimeException>> metadataErrors
+    ) {
+        AtomicReference<CompletableFuture<Map<TopicPartition, OffsetAndTimestampInternal>>> 
+                topicToOffsetsResult = new AtomicReference<>();
```

Comment:
I'm not following the need for this to be wrapped in an `AtomicReference`. All of the things happening are happening on the background thread, right? As such they don't need to be thread safe. (That's a major design goal of the new consumer.)

Reply:
I originally wrapped it in an additional layer to avoid the issue with Java lambdas requiring immutable variables. However, I’ve refactored this implementation to avoid any misunderstandings.